src/lv_misc/lv_txt_ap.c

```diff
@@ -0,0 +1,263 @@
+/**
+ * @file lv_txt_ap.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include <stddef.h>
+#include "lv_bidi.h"
+#include "lv_txt.h"
+#include "lv_conf.h"
+#include "lv_txt_ap.h"
+#include "../lv_draw/lv_draw.h"
+
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static uint32_t lv_ap_get_char_index(uint16_t c);
+#if LV_USE_REVERSE_ARABIC_PERSIAN_CHARS == 1
+static void lv_ap_normalize_chars(uint32_t * str,uint16_t count);
+#endif
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+#if LV_USE_ARABIC_PERSIAN_CHARS == 1
```

Comment:
- Yes, I forgot that.

- I did it to prevent error if `lv_txt_ap_calc_bytes_cnt` and `lv_txt_ap_proc` are used. But, yes, it should be on the entire file, because of if these functions are used, then `LV_USE_ARABIC_PERSIAN_CHARS` probably defined as `1`!